ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/PartyManagement.scala

```diff
@@ -183,4 +183,62 @@ final class PartyManagement(session: LedgerSession) extends LedgerTestSuite(sess
           s"The allocated party IDs $allocatedPartyIds are not a subset of $knownPartyIds.")
       }
   }
+
+  test(
+    "PMListKnowPartiesIsLocal",
+    "Listing parties on multi-node ledgers should reflect whether parties are local",
+    allocate(NoParties, NoParties),
+  ) {
+    case Participants(Participant(alpha), Participant(beta)) =>
+      for {
+        alphaLocalParty <- alpha.allocateParty(
+          partyIdHint = Some("PMListKnowPartiesIsLocal_" + Random.alphanumeric.take(10).mkString),
+          displayName = Some("Alice"),
+        )
+        betaLocalParty <- beta.allocateParty(
+          partyIdHint = Some("PMListKnowPartiesIsLocal_" + Random.alphanumeric.take(10).mkString),
+          displayName = Some("Bob"),
+        )
+        // only wait for parties on ledgers configured to know about each other's parties
+        _ <- alpha.waitForParties(Set(alpha, beta), Set(alphaLocalParty, betaLocalParty))
+        _ <- beta.waitForParties(Set(alpha, beta), Set(alphaLocalParty, betaLocalParty))
+        partyDetailsKnowToAlpha <- alpha.listKnownPartyDetails()
+        partyDetailsKnowToBeta <- beta.listKnownPartyDetails()
+      } yield {
+        def lookUp(
+            party: binding.Primitive.Party,
+            details: Seq[PartyDetails]): Option[(String, Boolean)] =
+          details.collectFirst {
+            case PartyDetails(p, displayName, isLocal)
+                if p == binding.Primitive.Party.unwrap(party) =>
+              (displayName, isLocal)
+          }
+
+        val Some((alice, aliceIsLocal)) = lookUp(alphaLocalParty, partyDetailsKnowToAlpha)
+        assert(alice == "Alice" && aliceIsLocal)
+
+        val Some((bob, bobIsLocal)) = lookUp(betaLocalParty, partyDetailsKnowToBeta)
+        assert(bob == "Bob" && bobIsLocal)
+
+        // On multi-participant ledgers remote parties should either not be known to each
+        // other or should be marked as non-local.
+        if (alpha.endpointId != beta.endpointId) {
+          lookUp(betaLocalParty, partyDetailsKnowToAlpha) match {
+            case None =>
+            case Some(("Bob", false)) =>
```

Comment:
There seems to be a bit of extra slack here. Why are these both acceptable behaviors?

Reply:
I added clarification in the comments in the code. Underlying this slack/freedom here is that distributed ledgers could choose to publish local parties to other participants or not.

I heard that privacy-aware KVUtils disabled the `waitForParties` hook because they did not have the assumption that remote parties were always visible to all participants.

Currently, Canton publishes parties to participants connected to the same domain by default, but even that could change based on configuration.